ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -194,6 +199,8 @@ public class DagUtils {
    */
   private final ConcurrentHashMap<String, Object> copyNotifiers = new ConcurrentHashMap<>();
 
+  private static final Cache<String, String> shaCache = CacheBuilder.newBuilder().maximumSize(100).build();
```

Comment:
nit: Consider using a Caffeine cache (more performant than Guava and has better utilities). Even though not the purpose of this PR. Maybe as part of another task.

Reply:
makes sense
we tend to use this gauva cache as far as I can see
if there is already a ticket about a migration, please mention here